packages/core/admin/admin/src/pages/ProfilePage/tests/index.test.js

```diff
@@ -54,23 +55,83 @@ describe('ADMIN | Pages | Profile page', () => {
   });
 
   afterAll(() => {
-    jest.resetAllMocks();
     server.close();
   });
 
-  it('renders and matches the snapshot', async () => {
-    const { container } = render(App);
+  it('renders and show the Interface Language section', async () => {
+    setup();
     await waitFor(() => {
       expect(screen.getByText('Interface language')).toBeInTheDocument();
     });
+  });
+
+  it('should display username if it exists', async () => {
+    setup();
+    await waitFor(() => {
+      expect(screen.getByText('yolo')).toBeInTheDocument();
+    });
+  });
+
+  it('should display the change password section and all its fields', async () => {
+    const { getByRole, getByTestId, queryByTestId } = setup();
+
+    await waitFor(() => {
+      expect(queryByTestId('loader')).not.toBeInTheDocument();
+    });
+
+    expect(
+      getByRole('heading', {
+        name: 'Change password',
+      })
+    ).toBeInTheDocument();
+
+    expect(getByTestId('test-current-password-input')).toBeInTheDocument();
+
+    expect(getByTestId('test-new-password-input')).toBeInTheDocument();
+
+    expect(getByTestId('test-confirmed-password-input')).toBeInTheDocument();
```

Comment:
while I do agree that in this case using the label would be better since it's available, but using testids has it's place and reasons it's definitely not "poor testing practice"

Reply:
In this case I think it definitely is, they're inputs the benefits of using `getByRole` or `getByLabelText` is that you're checking the accessibility of your UI as opposed to bipassing this.